x-pack/plugins/endpoint/common/types.ts

```diff
@@ -138,7 +142,7 @@ export interface LegacyEndpointEvent {
 }
 
 export interface EndpointEvent {
-  '@timestamp': Date;
+  '@timestamp': string;
```

Comment:
❔ Give the vagaries of dealing with time that you are especially familiar with, it might benefit the reader to have some comment describing what format is expected here.

Reply:
TBH I'm not sure "string" is right. Does anyone know how dates get serialized into JSON by default through `hapi`? The underlying value is a date equivalent in Elasticsearch, which according to the docs is stored as a 64-bit integer, so I would imagine that what we're more likely to see is that this is a `number` unless the `elasticsearch` node client/`hapi` combo is doing some `number`-->`Date`-->`string` conversion under the hood.

Then again, on second thought, this could be a `string | number` if we're working with the source document and just shoving it back to the client...